src/Microsoft.ML.Data/Scorers/PredictionTransformer.cs

```diff
@@ -0,0 +1,325 @@
+            Model = model;
+
+            // *** Binary format ***
+            // model: prediction model.
+            // stream: empty data view that contains train schema.
+            // id of string: feature column.
+            var ms = new MemoryStream();
+            ctx.TryLoadBinaryStream("TrainSchema", reader =>
+            {
+                reader.BaseStream.CopyTo(ms);
+            });
+
+            ms.Position = 0;
+            var loader = new BinaryLoader(host, new BinaryLoader.Arguments(), ms);
+            TrainSchema = loader.Schema;
+
+            FeatureColumn = ctx.LoadString();
+            TrainSchema.TryGetColumnIndex(FeatureColumn, out int col);
+            if (col < 0)
+                throw Host.ExceptSchemaMismatch(nameof(FeatureColumn), "feature", FeatureColumn);
+            FeatureColumnType = TrainSchema.GetColumnType(col);
+
+            BindableMapper = ScoreUtils.GetSchemaBindableMapper(Host, model);
+        }
+
+        public ISchema GetOutputSchema(ISchema inputSchema)
+        {
+            Host.CheckValue(inputSchema, nameof(inputSchema));
+
+            inputSchema.TryGetColumnIndex(FeatureColumn, out int col);
+            if (col < 0)
+                throw Host.ExceptSchemaMismatch(nameof(inputSchema), "feature", FeatureColumn, FeatureColumnType.ToString(), null);
+            if (!inputSchema.GetColumnType(col).Equals(FeatureColumnType))
+                throw Host.ExceptSchemaMismatch(nameof(inputSchema), "feature", FeatureColumn, FeatureColumnType.ToString(), inputSchema.GetColumnType(col).ToString());
+
+            return Transform(new EmptyDataView(Host, inputSchema)).Schema;
+        }
+
+        public abstract IDataView Transform(IDataView input);
+
+        public void Save(ModelSaveContext ctx)
+        {
+            Host.CheckValue(ctx, nameof(ctx));
+            ctx.CheckAtModel();
+            SaveCore(ctx);
+        }
+
+        protected virtual void SaveCore(ModelSaveContext ctx)
+        {
+            // *** Binary format ***
+            // model: prediction model.
+            // stream: empty data view that contains train schema.
+            // id of string: feature column.
+
+            ctx.SaveModel(Model, "Model");
+            ctx.SaveBinaryStream("TrainSchema", writer =>
+            {
+                using (var ch = Host.Start("Saving train schema"))
+                {
+                    var saver = new BinarySaver(Host, new BinarySaver.Arguments { Silent = true });
+                    DataSaverUtils.SaveDataView(ch, saver, new EmptyDataView(Host, TrainSchema), writer.BaseStream);
+                }
+            });
+
+            ctx.SaveString(FeatureColumn);
+        }
+    }
+
+    public sealed class BinaryPredictionTransformer<TModel> : PredictionTransformerBase<TModel>
+    where TModel : class, IPredictorProducing<float>
+    {
+        private readonly BinaryClassifierScorer _scorer;
+
```

Comment:
I suppose in future iterations of this we will get rid of these so this "prediction transform" isn't just a shim over scorers, but is the only method, I suppose. Maybe.

Reply:
Yes, these guys need to *become* scorers, but in a separate review

---
In reply to: [212449296](https://github.com/dotnet/machinelearning/pull/716#discussion_r212449296) [](ancestors = 212449296)